web/custom-plugins/forecast-of-response/src/components/Result.tsx

```diff
@@ -96,79 +144,67 @@ function GraphSelect(props: {
       } else {
         newDataPoint = {
           timestamp: timestamp,
-          [plotName]: result.values[index],
+          [graphName]: result.values[index],
           ...newDataDict[timestamp],
         }
       }
       newDataDict[timestamp] = newDataPoint
     })
 
-    setGraphInfo([
-      ...graphInfo,
+    return [
       {
-        name: plotName,
+        name: graphName,
         plotType: result?.plotType,
         unit: result?.unit,
         description: description,
+        uuid: uuid
       },
-    ])
-    setChartData(Object.values(newDataDict))
+      newDataDict
+    ]
   }
 
   return (
     <GraphSelectorWrapper>
-      <StyledSelect onChange={(e: Event) => setRun(e.target.value)} value={run}>
+      <StyledSelect onChange={(e: Event) => setChosenRun(e.target.value)}>
         {variableRuns.map((run: any, index) => (
           <option key={index} value={index}>
             {run.name}
           </option>
         ))}
       </StyledSelect>
-      {variableRuns[run] && (
-        <StyledSelect
-          onChange={(e: Event) => setResponse(e.target.value)}
-          value={response}
-        >
-          {variableRuns[run].responses.map((response: any, index: number) => (
+      <StyledSelect onChange={(e: Event) => setChosenResponse(e.target.value)}>
+        {variableRuns[chosenRun].responses.map((response: any, index: number) => (
+          <option key={index} value={index}>
+            {response.name}
+          </option>
+        ))}
+      </StyledSelect>
+      <StyledSelect onChange={(e: Event) => setChosenStatistic(e.target.value)}>
+        {variableRuns[chosenRun].responses[chosenResponse].statistics.map(
+          (statistic: any, index: number) => (
             <option key={index} value={index}>
-              {response.name}
+              {statistic.name}
             </option>
-          ))}
-        </StyledSelect>
-      )}
-      {variableRuns[run].responses[response] && (
-        <StyledSelect
-          onChange={(e: Event) => setStatistic(e.target.value)}
-          value={statistic}
-        >
-          {variableRuns[run].responses[response].statistics.map(
-            (statistic: any, index: number) => (
-              <option key={index} value={index}>
-                {statistic.name}
-              </option>
-            )
-          )}
-        </StyledSelect>
-      )}
+          )
+        )}
+      </StyledSelect>
       <Button
         style={{ width: '140px', marginLeft: '10px' }}
-        onClick={() => addGraph()}
+        onClick={() => {addGraphFromSelector()}}
       >
         Add graph
         <Icons name="add" title="add" />
       </Button>
-      {index ? (
+      {isRootPlot ? (
```

Comment:
```suggestion
      {!isRootPlot && <ReturnWhatIWantToSeeOnly />}
```
Better than rendering empty divs ;)